### PR TITLE
Notify users when a shipped default moves under their customized prompt

### PR DESCRIPTION
**Pull request by Leo310**
Follow-up to #402 — the last gap in #401's notification story.

## The gap

Skills and prompts disagreed about the customized case:

| | untouched | customized |
|---|---|---|
| **Skill** | silently updated | notice + Review |
| **Prompt** | silently updated | **nothing** |

`detectStaleGuidance` skipped `version === null` outright, so a user who edited their base or memory prompt was **never told** when we later shipped an improved default. That silently excluded exactly the users engaged enough to customize the prompt in the first place — the improvement was invisible to them unless they happened to open the Agent editor and spot the "Diff with default" button.

## Why the skip wasn't arbitrary

A customized file matches no shipped fingerprint, so the content alone cannot distinguish:

- "you edited it **and the default has since moved**" → worth a notice
- "you edited it **and nothing changed**" → silence

Both are just `null`. Removing the skip would have fired a false "the default changed" notice at every user who has ever customized a prompt.

## The fix

Adds `AgentConfig.promptBaseVersions` — which shipped version each prompt file was last written from, keyed by prompt kind. Stamped on **every** write: seeding, the silent old-default update, a reset, and the user saving their own text (which baselines their edit at today's default, so a *later* bump is what raises the notice).

Drift then becomes exact: **stamp exists AND stamp ≠ current**.

Edge cases, all pinned by tests:
- **No stamp** → stays silent. We can't substantiate drift we have no baseline for.
- **Customized file predating the field** → gets a stamp seeded at the current version, so it's detectable from the next bump onward. Deliberately *not* backdated — that would claim their edit predates a change it may already incorporate.
- **Existing older stamp** → never overwritten, since that would erase the very drift the field exists to detect.

Stamping is non-fatal by design: it's bookkeeping, and a store hiccup must not abort the surrounding write and leave the user with no prompt file at all. Worst case, a stamp is missed and drift stays silent.

## Scope

Only the detection was missing — the notice, the `customized: true` wording ("your customized version was kept"), the version-stamped dismissal id, and Review-opens-the-diff all already handle this case from #402.

One incidental fix: `test/agent/promptFiles.test.ts` mocked `getData()` without `updateAgent`, which the stamp now calls. That surfaced the non-fatal concern above, which is why stamping is wrapped rather than left to throw.

## Verification

`bun run test` — 1380 passing (82 files, up from 1359/81), `bun run check` clean, Biome clean on changed files, `bun run build` succeeds.

New coverage: the stamp is written on seed / silent-update / user-save / back-fill and never overwritten when older; and on the detection side, a superseded baseline flags with `customized: true` while a current baseline and an absent baseline both stay silent.

_AI assistance: written with AI coding agents from the maintainer's brief; reviewed and tested by the maintainer._

### CODE REVIEW

File: src/agent/promptFiles.ts [hunk 1]
```diff
@@ -252,6 +270,49 @@ export class PromptFilesService {
 		await this.ensureParent(path);
 		await this.adapter.write(path, text);
 		this.cache(kind).set(agentId, text);
+		this.stampBaseVersion(kind, agentId);
```
[the remark this answers is not in the document]
Comment (Leo310) [reply src/agent/promptFiles.ts:280]: Valid — fixed in ece7bb2. `copyAgentPrompts` now inherits the source's baseline after the write (and clears the stamp when the source has none, rather than leaving `write`'s current-version stamp), so duplicating an agent no longer hides its pending drift notice; both directions are pinned by tests that fail without the fix.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_
